onyo/lib/onyo.py

```diff
@@ -900,3 +900,46 @@ def _rm_sanitize(self, paths: Iterable[Union[Path, str]]) -> list[Path]:
                                          '\n'.join(error_path_protected))
 
         return paths_to_rm
+
+    def validate_name_scheme(self, name: str) -> bool:
```

Comment:
IMO, this belongs in the "NEW" section.

Also `validate_name_scheme()` -> `valid_name()`

is there a reason why you felt that `re` wasn't the right approach here? To me it feels like a better fit, and would greatly simplify the function.